### PR TITLE
Added INSPIRE HEP Fetcher to "Update with bibliographic information from the web"

### LINKED ISSUE #9351: Add INSPIRE HEP Fetcher to "Update with bibliographic information from the web"
**Is your suggestion for improvement related to a problem? Please describe.**
https://discourse.jabref.org/t/add-inspirehep-as-resource-to-update-the-bibliographic-data/3612

However, it would be nice to add InspireHEP as one of the resources to fetch bibliographic data from, so these keys can also be fetched from InspireHEP for entries, which had been imported from elsewhere or are missing an citation key.

**Describe the solution you'd like**
Add the fetcher to the LIST

**Additional context**

grafik

**Implementation details**
The INSPIRE HEP Fetcher needs to implement EntryBasedFetcher interface and needs to be added to the set of getEntryBasedFetchers in WebFetcher.class

The fields doi or arxivid could be used as fields
https://github.com/inspirehep/rest-api-doc#external-identifiers

### PR DESCRIPTION
**Pull request by lak7**
Resolves #9351 

Added INSPIREFetcher to the set of GetEntryBasedFetchers and also updated INSPIREFetcher to implement EntryBaseFetcher Interface.

Screenshot_20230207_181029

This adds the Inspire button where one can fetch the data for existing entries.

TEST CASE:
Here for example is the doi for testing Inspire Fetcher: 10.1103/PhysRevLett.126.101101
Now we can fetch bibliographic data from InspireHEP for entries, which may had been imported from elsewhere or are missing an citation key. 
Screenshot_20230214_235455

- [ ] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [x ] Manually tested changed features in running JabRef (always required)
- [x ] Screenshots added in PR description (for UI changes)
- [ ] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

File: src/main/java/org/jabref/logic/importer/fetcher/INSPIREFetcher.java [hunk 1]
```diff
@@ -78,4 +104,42 @@ public void doPostCleanup(BibEntry entry) {
...
+
+        String doi = entry.getField(StandardField.DOI).orElse(null);
+        if (doi == null) {
+            return results;
+        }
+
+        String url = "https://inspirehep.net/api/doi" + doi;
+
+        try {
+            URL obj = new URL(url);
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/logic/importer/fetcher/INSPIREFetcher.java:120]: You can  use our UrlDownload class for this
  
   URLDownload download = getUrlDownload(doiURL);
                download.addHeader("Accept", MediaTypes.APPLICATION_BIBTEX);
                
     This also handles errors

**Review by Siedlerchr — CHANGES_REQUESTED** [review]
Thanks for your contribution. Goes already in the right direction, but something seems to be off with your formatting style. Please import JabRef's code style

 https://github.com/JabRef/jabref/tree/main/config#intellij-idea-code-style-configuration
 
 And also please add a test case!

File: src/main/java/org/jabref/logic/importer/fetcher/INSPIREFetcher.java (hunk 1 above)
Comment (lak7) [reply src/main/java/org/jabref/logic/importer/fetcher/INSPIREFetcher.java:99]: @Siedlerchr I have fixed some logical bugs and corrected code style.

Now InspireHEP is working as desired as one of the resources to fetch bibliographic data from when given a valid doi.

Screenshot_20230214_235455

File: src/main/java/org/jabref/logic/importer/fetcher/INSPIREFetcher.java [hunk 2]
```diff
@@ -78,4 +82,43 @@ public void doPostCleanup(BibEntry entry) {
...
+        if (doi == null) {
+            return results;
+        }
+
+        String url = INSPIRE_EXTERNAL_HOST + doi;
+
+        try {
+
+            URL obj = new URL(url);
+            URLDownload download = getUrlDownload(obj);
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/logic/importer/fetcher/INSPIREFetcher.java:100]: You can simplify this even more: Just download.asString() gives you the resoure as string.

File: src/main/java/org/jabref/logic/importer/fetcher/INSPIREFetcher.java [hunk 3]
```diff
@@ -78,4 +82,43 @@ public void doPostCleanup(BibEntry entry) {
     public Parser getParser() {
         return new BibtexParser(importFormatPreferences, new DummyFileUpdateMonitor());
     }
+
+    @Override
+    public List<BibEntry> performSearch(BibEntry entry) {
+        List<BibEntry> results = new ArrayList<>();
+        String doi = entry.getField(StandardField.DOI).orElse(null);
+
+        if (doi == null) {
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/logic/importer/fetcher/INSPIREFetcher.java:91]: Optional already provides methods like isPresent ...to check if the optional has a value. And the purpose of Optional is to avod nulls
https://www.baeldung.com/java-optional

**Review by Siedlerchr — CHANGES_REQUESTED** [review]

**Review by calixtus — APPROVED** [review]

**Pull request merged**